### DISCUSSION TITLE
Welcome to SCTE-35-HLS-ABR_umzz Discussions!

### DISCUSSION THREAD

**Discussion by futzu** [futzu/umzz#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gauravv5** [comment]
HI Futzu..!!
Hope you are doing well.
if I encode one udp stream to multi bitrate using ffmpeg. Does UMZZ still work. as we have discussed earlier in x9k3 discussion that ffmpeg changes data(scte25)(0x86) to data-bin(0x06) pid.

Thanks and hope this is taken care in this umzz project

**Comment by futzu** [comment]
Yeah, threefive does the SCTE-35 parsing and it parses 0x86 and 0x06 for SCTE-35.

If you can figure out how to multi-bitrate  without ffmpeg stripping the data  streams out.
the var_stream_map thing only allows audio,video, and subtitles. 

If you figure it out, let me know.

**Comment by futzu** [comment]
You can also  parse the SCTE-35 out and write it to a sidecar file and pipe the video to ffmpeg like this 

* proxy.py
```py3
import sys
from threefive import Stream

def mk_sidecar(cue):
    with open("sidecar.txt","a") as sidecar:
            data = f'{cue.packet_data.pts},{cue.encode()}\n'
            sidecar.write(data)


if __name__=='__main__':
    stream = Stream(sys.argv[1])
    stream.proxy(func=mk_sidecar)

```
* Stream.proxy writes to stdout , you can pipe it into ffmpeg
```py3
python3 proxy.py udp://example.com/video.ts | ffmpeg -i - {ffmpeg stuff} -o master.m3u8
```
* have umzz use the sidecar file 
```py3
umzz -i master.m3u8 -s sidecar.txt
```

**Comment by abhi4100** [comment]
If I am not use sidecar file but i have scte 35 in my segments. so can i add a tag in master m3u8

**Reply by futzu** [reply]
I'm not entirely sure what it is your asking, 
but I believe you want to know if SCTE-35 Cues in your segments will  be converted into HLS tags, 
<br>

__the answer is yes,  SCTE-35 Cues in Mpegts Streams are Translated into HLS tags.__

<br> umzz is mostly multi-process [x9k3](https://github.com/futzu/x9k3), 
you might want to read the x9k3 docs, and it will give you a better idea of how things work.
